### PR TITLE
p_nom and s_nom after final iteration of network_lopf

### PR DESCRIPTION
**Pull request by eb5194**
In `pypsa.linopf`, the last iteration of `network_lopf`, the nominal power of lines and links was set to previous optimal value, which should be changes back after the last optimisation.

I also adapted the release notes from a previous PR as it was incorrect and is sort of related to the last network_lopf iteration

EDIT: I don't know why previous changes re-appear (track_iterations of global constraints, but it's just one line of code). I thought they already were merged into the master repo?

EDIT2: Dunno what I did to erase it, but it disappeared. Yay.

### CODE REVIEW

File: doc/release_notes.rst [hunk 1]
```diff
@@ -31,6 +31,12 @@ This release contains bug fixes and extensions to the features for optimization
 
 * When solving ``n.lopf(pyomo=False, store_basis=True, solver_name="cplex")`` an error raised by trying to store a non-existing basis is caught.
 
+* Added track iterations of global constraints
+
+* overwrite ``p_nom`` and ``s_nom`` of line and link attributes to original values after final optimisation
+
+* When solving ``n.lopf(pyomo=False)``, PyPSA now constrains the dispatch variables for non extendable components with actual constraints, not with standard variable bounds. This allows retrieving shadow prices for all dispatch variables when running ``n.lopf(pyomo=False, keep_shadowprices=True).   
+
```
Comment (fneum) [inline doc/release_notes.rst:39]: There was a new release on Wednesday and the release notes for your pull request should go above this release (v0.17.1). Also "When solving..." is now duplicated in the release notes.

File: pypsa/linopf.py [hunk 2]
```diff
@@ -1035,9 +1035,11 @@ def save_optimal_capacities(n, iteration, status):
     logger.info('Running last lopf with fixed branches, overwrite p_nom '
                 'for links and s_nom for lines')
     ext_links_i = get_extendable_i(n, 'Link')
+    original_lnv = n.lines.s_nom.copy()
+    original_lkv = n.links.p_nom.copy()
```
Comment (fneum) [inline pypsa/linopf.py:1039]: maybe rename `original_lnv` to `s_nom_orig`? This would fit better with the attribute name.

File: pypsa/linopf.py [hunk 3]
```diff
@@ -1035,9 +1035,11 @@ def save_optimal_capacities(n, iteration, status):
...
     n.lines[['s_nom', 's_nom_extendable']] = n.lines['s_nom_opt'], False
     n.links[['p_nom', 'p_nom_extendable']] = n.links['p_nom_opt'], False
     kwargs['warmstart'] = False
     network_lopf(n, snapshots, **kwargs)
-    n.lines.loc[ext_i, 's_nom_extendable'] = True
-    n.links.loc[ext_links_i, 'p_nom_extendable'] = True
+    n.lines.loc[ext_i, ['s_nom', 's_nom_extendable']] = original_lnv.loc[ext_i], True
+    n.links.loc[ext_links_i, ['p_nom', 'p_nom_extendable']] = original_lkv.loc[ext_links_i], True
```
Comment (fneum) [inline pypsa/linopf.py:1045]:
```suggestion
    n.lines.loc[ext_i, ['s_nom', 's_nom_extendable']] = original_lnv[ext_i], True
    n.links.loc[ext_links_i, ['p_nom', 'p_nom_extendable']] = original_lkv[ext_links_i], True
```
This should also work.

**Comment by nworbmot** [comment]
Let's discuss this on Thursday before merging. I'm not quite sure without a deep dive what the issues are.

**Comment by eb5194** [comment]
> Hey @eb5194 , I'm not sure if we should do this. In a normal setup resistance and impedance parameters refer to the installed `s_nom`. If we reset the `s_nom` to the original value, we loose this relation, and the parameter "only" refer to `s_nom_opt`. perhaps @nworbmot has a opinion about that? (perhaps in the long term it would be even better to introduce a `x_opt`, `r_opt`?)

We discussed it in the group-meeting on Thursday and agreed to change it as I proposed. It simply makes more sense to keep the old values instead of always having `s_nom=s_nom_opt` and the same for `p_nom=p_nom_opt`. All parameters of reseistance etc. then refer to the optimal values and in the same time, you can see which line was reinforced in the optimisation process.

**Comment by eb5194** [comment]
hups, forgot about this pr, but should be fine now.

**Review by fneum — APPROVED** [review]

**Pull request merged**